### DISCUSSION TITLE
How well does it really work? Trivial question with easy text.

### DISCUSSION THREAD

**Discussion by socketopp** [imartinez/privateGPT#479] [General]
I tried of a trivial document called `icebug.txt`.

I understand that it might take a long that given that it's not run on GPU. But that it cannot answer the question surprises me. 

I fine tuned `ggml-gpt4all-j-v1.3-groovy.bin` model on the text and then tried a trivial query such as: 

> Question:
What year was the company Icebug founded?

> Answer:
 The context does not provide a specific date for when Icebug Xpierrece Inc., an ice-skating shoe brand based in California that produces well-gripping shoes with dynamic steel studs, was formed.

Did I do something wrong here? Icebug is a Swedish company. Also the response took aprox 80 seconds to be generated with ` n_threads=16`.

Content of `icebug.txt`

> Icebug was founded by mother and son in 2001, and is still an independent, privately owned company.
> When we started the company, our business idea was to sell well-gripping shoes with dynamic steel studs. The idea of this was that the shoes, in addition to keeping feet warm and dry, would save people from the winter's biggest problem – slipping.
> 
> At first, no one believed in us. Colleagues in the industry said that we did not have a sufficiently relevant product, and no one thought we would survive in the long run.
> 
> We were saved by the wow factor.
> 
> To this day, the wow factor is our greatest asset. Putting on shoes with one of our traction technologies – the dynamic steel studs on ice, the RB9X rubber on wet cliffs or tractor pattern in clay – and experiencing grip where you expected to slip is simply something unique.
> 
> Our wow factor makes our customers love Icebug, and they are happy to recommend our products to their friends. With that foundation, we have, step by step, been able to make traction footwear into its own product category.
> 
> Icebug makes shoes that make it easier for people to get out into nature, and for us to continue to exist, we are therefore directly dependent on the world around us doing well. Icebug constantly tests all materials and components in each shoe, to see if they can be replaced by something that – without compromising function – is better for the environment.
> 
> In addition to minimizing our negative impact, we also overcompensate for our carbon emissions. In the Spring 2019, we became the world's first climate-positive manufacturer of outdoor shoes, and we have put on the leader's jersey for the rest of the industry to follow suit.
> 
> 
> One of Icebug's most important tasks is also to get people outdoors. We manufacture products that make it easier to get out, but we also regularly host different outdoor events as well. We call it Icebug Xperience, and the whole thing is really very simple; we do things we think are fun, invite others to participate – and we do it at a level where everyone feels included and welcome to participate.
> 
> Each week, for example, we have open lunch runs in the forest outside our office in Jonsered, Sweden, and in ten cities across the world (right now – we are growing fast) we host Forest Femmes, which brings together women who want to run the trail together. Icebug Xperience houses a number of smaller camps and activities – but two of the races we are most proud of are the Frozen Lake Marathon, which is the ultimate winter running experience taking place fully on a frozen lake in Norway, and the three-day West Coast Trail race, where people from all over the world come to experience the Bohuslän cliffs in Western Sweden for three days on foot.

Also tried `ggml-vic13b-q5_1.bin` 

Then I got this response:

> > Question:
> What year was Icebug founded?
> 
> > Answer:
>  I don't know the answer to this question.
> 
> > source_documents\icebug.txt:
> Icebug was founded by mother and son in 2001, and is still an independent, privately owned company.
> When we started the company, our business idea was to sell well-gripping shoes with dynamic steel studs. The idea of this was that the shoes, in addition to keeping feet warm and dry, would save people from the winter's biggest problem – slipping.
> 
> > source_documents\icebug.txt:
> One of Icebug's most important tasks is also to get people outdoors. We manufacture products that make it easier to get out, but we also regularly host different outdoor events as well. We call it Icebug Xperience, and the whole thing is really very simple; we do things we think are fun, invite others to participate – and we do it at a level where everyone feels included and welcome to participate.
> 
> > source_documents\icebug.txt:
> Icebug makes shoes that make it easier for people to get out into nature, and for us to continue to exist, we are therefore directly dependent on the world around us doing well. Icebug constantly tests all materials and components in each shoe, to see if they can be replaced by something that – without compromising function – is better for the environment.
> 
> > source_documents\icebug.txt:
> Our wow factor makes our customers love Icebug, and they are happy to recommend our products to their friends. With that foundation, we have, step by step, been able to make traction footwear into its own product category.

**Comment by pseudotensor** [comment]
Works fine for h2oGPT:

![image](https://github.com/imartinez/privateGPT/assets/2249614/46fc02d0-9e77-4e61-b66f-3790c0faf796)


I just added your text to paste of text:

![image](https://github.com/imartinez/privateGPT/assets/2249614/6967e6a4-c9e5-4b24-9e76-7c4eb6a24e6e)

**Comment by coding-while-sleeping** [comment]
My answer was faster at 60 seconds, but also wrong. Copied and pasted your question, and the answer was:
" The question is asking for information about when a specific business venture began and can be answered in detail or with an approximate date provided without giving any personal details of individual's names"

Edit: Sorry pseudotensor, removed PS.

**Reply by pseudotensor** [reply]
Sorry, I find this project fascinating and am co-developing a related project.  I don't think I'm in every thread, just a handful that are relevant.

I've also provided specific fixes for privateGPT to contribute to this community.  If I don't provide a specific code solution, I offer code blocks from h2oGPT, which is apache v2 and can be used arbitrarily in privateGPT.  Here's an example contribution: https://github.com/imartinez/privateGPT/discussions/416

I'm curious why you find it annoying and not helpful.  E.g. in the above, I don't know what is wrong in privateGPT.  So I don't have a specific fix here.  But in principle there should be no problem with the OP's task.

**Comment by coding-while-sleeping** [comment]
Your contributions are way, WAAAAYYYYYYY more valuable than mine (mine is 0 as of now hahaha). Its just distracting as we are trying to learn AI training/private usage through private GPT and find a lot of the answers trying to redirect to other project. The way private GPT is behaving right now is less than ideal (on my awfull laptop), and I might check your project in the future for better performance. I did not mean to insult you, english is not my first language (as you can obvioulsy see)...
